File: leetcode/151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        l=s.split()
        str= ""
        for i in range(len(l)-1, -1, -1) :
            str += l[i]
            str += " "
        return str[0:len(str)-1]
            

File: leetcode/151-reverse-words-in-a-string/reverse-words-in-a-string.cpp
class Solution {
public:
    string reverse(string str){
        int st=0, end=str.length()-1;
        while(st<=end){
            swap(str[st],str[end]);
            st++;
            end--;
        }
        return str;
    }
    string reverseWords(string s) {
        string str= reverse(s);
        string str2;
        for(int i=0; i<str.length(); i++){
            string w;
            while(i<str.length() && str[i]!=' '){
                w=w+str[i];
                i++;
            }
            if (w.length()>0){
                str2+=" " + reverse(w);
            }
        }
        return str2.substr(1);
    }
};auto init=atexit([]{ofstream("display_runtime.txt")<<"0";});